tip-router-operator-cli/src/lib.rs

```diff
@@ -113,53 +132,128 @@ fn emit_inconsistent_tree_node_amount_dp(
-                format!("tip_distribution_account={tip_distribution_account},actual_claims={actual_claims}, expected_claims={expected_claims}"),
-                String
-            ),
+            ("actual_claims_exceeded", format!(
+                "tip_distribution_account={tip_distribution_account},actual_claims={actual_claims}, expected_claims={expected_claims}"
+            ), String)
         );
     } else {
-        datapoint_warn!(
-            "tip-distributor",
-            (
-                "actual_claims_below",
-                format!("tip_distribution_account={tip_distribution_account},actual_claims={actual_claims}, expected_claims={expected_claims}"),
-                String
+        datapoint_warn!("tip-distributor", (
+            "actual_claims_below",
+            format!(
+                "tip_distribution_account={tip_distribution_account},actual_claims={actual_claims}, expected_claims={expected_claims}"
             ),
-        );
+            String
+        ));
     }
 }
 
 impl GeneratedMerkleTreeCollection {
-    pub fn new_from_stake_meta_collection(
+    pub async fn new_from_stake_meta_collection(
         stake_meta_coll: StakeMetaCollection,
-        maybe_rpc_client: Option<SyncRpcClient>,
+        protocol_fee_bps: u16
     ) -> Result<GeneratedMerkleTreeCollection, MerkleRootGeneratorError> {
-        let generated_merkle_trees = stake_meta_coll
-            .stake_metas
+        let (config_pda, _) = Pubkey::find_program_address(
+            &[CONFIG_ACCOUNT_SEED],
+            &stake_meta_coll.tip_distribution_program_id
+        );
+
+        let generated_merkle_trees = stake_meta_coll.stake_metas
             .into_iter()
             .filter(|stake_meta| stake_meta.maybe_tip_distribution_meta.is_some())
-            .filter_map(|stake_meta| {
-                let mut tree_nodes = match TreeNode::vec_from_stake_meta(&stake_meta) {
-                    Err(e) => return Some(Err(e)),
-                    Ok(maybe_tree_nodes) => maybe_tree_nodes,
-                }?;
-
-                if let Some(rpc_client) = &maybe_rpc_client {
-                    if let Some(tda) = stake_meta.maybe_tip_distribution_meta.as_ref() {
-                        emit_inconsistent_tree_node_amount_dp(
-                            &tree_nodes[..],
-                            &tda.tip_distribution_pubkey,
-                            rpc_client,
-                        );
-                    }
-                }
+            .map(|stake_meta| {
+                let tip_distribution_meta = stake_meta.maybe_tip_distribution_meta
+                    .as_ref()
+                    .unwrap();
 
-                let hashed_nodes: Vec<[u8; 32]> =
-                    tree_nodes.iter().map(|n| n.hash().to_bytes()).collect();
+                // Calculate fees
+                let protocol_fee_amount = (((tip_distribution_meta.total_tips as u128) *
+                    (protocol_fee_bps as u128)) /
```

Comment:
probably should use `div_ceil` when calculating protocol_fee_amount. Also better to use checked math to handle overflow errors